### ISSUE TITLE
Cannot change polymorhpic type on existing record

### ISSUE THREAD

**Issue by davorbadrov** [mathieuprog/polymorphic_embed#29]
Hi, great lib, it really helps on the project I'm working on.

I'm having a problem with changing the embedded type on a struct: when I create a struct with one polymorphic type and then when I try to change to other type it fails, or more specific for my situation - I create an entry with an email reminder and then I want to change it to an SMS reminder, but when I save it it remains as Email. Could you help me out please?

I wrote a failing test based on your existing tests to help you figure out what I'm trying to achieve. The changeset doesn't register the change channel change, only date and text. Is this a bug or am I just doing something wrong?

```elixir
test "changing type" do
    reminder_module = get_module(Reminder, true)
    email_module = get_module(Channel.Email, true)
    sms_module = get_module(Channel.SMS, true)

    attrs = %{
      date: ~U[2020-05-28 02:57:19Z],
      text: "This is an Email reminder",
      channel: %{
        address: "john@example.com",
        valid: true,
        confirmed: false
      }
    }

    insert_result =
      struct(reminder_module)
      |> reminder_module.changeset(attrs)
      |> Repo.insert()

    assert {:ok, %reminder_module{} = reminder} = insert_result

    update_attrs = %{
      date: ~U[2020-05-29 02:57:19Z],
      text: "This is an SMS reminder",
      channel: %{
        my_type_field: "sms",
        number: "02/807.05.53",
        country_code: 1,
        attempts: [],
        provider: nil
      }
    }

    update_result = reminder
    |> reminder_module.changeset(update_attrs)
    |> Repo.update()

    assert {:ok, %reminder_module{} = updated_reminder} = update_result

    reminder =
      reminder_module
      |> QueryBuilder.where(text: "This is an SMS reminder")
      |> Repo.one()

    # I expected this to be true
    assert sms_module == reminder.channel.__struct__
  end
```

**Comment by mathieuprog** [comment]
Thank you for adding a failing test. It's quite a significant bug so I'll try to resolve it soon.

**Comment by davorbadrov** [comment]
That sound great, thanks and keep up the great work :+1:

**Comment by mathieuprog** [comment]
This has been fixed in 1.3.3
